numpy/core/tests/test_multiarray.py

```diff
@@ -1748,6 +1748,32 @@ def test_transpose(self):
         assert_raises(ValueError, lambda: a.transpose(0, 0))
         assert_raises(ValueError, lambda: a.transpose(0, 1, 2))
 
+    def test_sort_by(self):
+        a = np.array([-2, -3, -4, -1, -6])
+        b = np.sort(a, by=(np.absolute(a),))
+        e = np.array([-1, -2, -3, -4, -6])
+        assert_equal(b, e)
+        # Test sorting complex numbers by absolute value
+        carr = np.arange(4, dtype=np.complex128)[::-1].reshape(2, 2)
+        carr_out = np.array([[2+0j, 3+0j], [0+0j, 1+0j]], dtype=np.complex128)
+        carr.sort(axis=1, by=(np.absolute(carr),))
+        assert_equal(carr, carr_out)
+        #Testing sort by real, imag
+        carr = np.array([1 + 1j, 1-1j], dtype=np.complex128)
+        carr_out = np.array([1-1j, 1+1j], dtype=np.complex128)
+        carr.sort(by=(carr.imag, carr.real))
+        assert_equal(carr, carr_out)
+        # Look for different shape error
+        carr = np.array([[1+1j, 1-1j]], dtype=np.complex128)
+        assert_raises(ValueError, carr.sort, by=(carr.imag[0],))
+        # Only stable sort test
+        carr = np.array([[1+1j, 1-1j]], dtype=np.complex128)
+        assert_raises(ValueError, carr.sort, kind='quicksort')
```

Comment:
This test should be transformed into one using `by=`, lets not modify the complex right now.